### PR TITLE
feat: no_superfluous_phpdoc_tags - also cover ?type

### PR DESCRIPTION
**Pull request by keradus**
closes https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/discussions/8110

### CODE REVIEW

File: src/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixer.php [hunk 1]
```diff
@@ -637,6 +637,13 @@ private function annotationIsSuperfluous(
      */
     private function toComparableNames(array $types, ?string $namespace, ?string $currentSymbol, array $symbolShortNames): array
     {
+        if (1 === \count($types) && '?' === $types[0][0]) {
```
Comment (kubawerlos) [inline src/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixer.php:640]:
```suggestion
        if ('?' === $types[0][0]) {
```
Checking for `?` as the first character is enough, it is not possible to mix `?` with the union type.

The first mutation worries me, why mutant with the change:
```diff
- if (1 === \count($types) && '?' === $types[0][0]) { 
+ if (1 !== \count($types) && '?' === $types[0][0]) {
```
escapes? Or it is not, and something related to Infection is not working.

Comment (keradus) [reply src/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixer.php:640]: `?` check only - with current PHP yes. in future, maybe `?foo|bar` syntax would be allowed (like alias for `null|foo|bar`). dunno. but I know I won't remember to update the code, so I prefer to be double safe. no strong opinion on it.

infection - i deep dive. `toComparableNames` is called too many times, let's see if it improved for this PR after https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8132 is merged

File: tests/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixerTest.php [hunk 2]
```diff
@@ -1519,6 +1519,25 @@ public function doFoo(?Bar $bar): ?Baz {}
...
+    public function doFoo(?Bar $bar): ?Baz {}
+}',
+            '<?php
+class Foo {
+    /**
+     * @param ?Bar $bar
+     *
+     * @return ?Baz
+     */
+    public function doFoo(?Bar $bar): ?Baz {}
```
Comment (Wirone) [inline tests/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixerTest.php:1537]: We probably need also a test case with:

```
     * @param ?Bar $bar
     *
     * @return ?Baz
     */
    public function doFoo(Bar|null $bar): ?Baz {}
```

Comment (keradus) [reply tests/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixerTest.php:1537]: I think they were never added for 8.0 support

**Review by kubawerlos — APPROVED** [review]

**Pull request merged**